### DISCUSSION TITLE
direct simulation of magnetic flux density function

### DISCUSSION THREAD

**Discussion by jahn234** [simnibs/simnibs#215] [8. Simulation workflows]
Hello everyone, I need to directly simulate the magnetic flux density and electric field function intensity inside the head for a scientific project. My desired function should be within a circular region with a radius slightly larger than the size of a human skull. Could anyone help me for this task? I really need this help. thanks

**Comment by oulap** [comment]
Try running the simulation and writing out both the E- and J-fields. 

If you're using the GUI, see "Setting simulation options" here:

https://simnibs.github.io/simnibs/build/html/tutorial/gui.html

If you're scripting the simulations you can choose which fields get written out by adding this line (in python):

`s.fields = 'eEjJ'`

where s is defined as:

`s = sim_struct.SESSION()`

the 'e' field that gets written into the simulation output mesh, is the electric field magnitude, which might be what you refer to as "electric field functionn intensity". For the magnetic field B, you can use this command to turn the J into B:

https://simnibs.github.io/simnibs/build/html/documentation/command_line/calc_b.html

but the flux density you need to compute yourself from the B given a surface.

**Reply by jahn234** [reply]
I appreciate your explanations, but it seems I couldn't express my work concept clearly. I want to simulate a function (Like B=x^2+y+z in cartesian) as my own flux density function inside the head. In this way, I aim to simulate the electric field inside the head without using coils or electrodes. I only want to utilize the tissue properties of the brain and its electrical conductivity in each tetrahedron to demonstrate the strength of the electric field (E-field) based on the magnetic flux density.
